File: bubbleSort.pl
#!/usr/bin/perl

use strict;
use warnings;

sub bubbleSort{
        print "Unsorted list:  @_ \n";
        my $swap=1; #swap is true

        while($swap){ #while swap is true
                $swap = 0; #swap is false / unsorted
                for(my $i = 1; $i < @_; $i++){
                        if($_[$i] < $_[$i-1]){
                                $swap=1; #swapping is true / sorting
                                @_[$i,$i-1]=@_[$i-1,$i];
                        }
                }
        }
        return @_;
}

my @arr = (10,-34,119,0,22,190,10500,-22,-65,-1);
bubbleSort(@arr);
print "Sorted list: @arr \n";

File: bubbleSort.py
#!/usr/bin/python
#Python implementation of Bubble Sort

def bubbleSort(list):
        print("Unsorted list: ",list)
        for x in range(len(list)-1): #range 0 to length of array (10)-1 = range(9)
                for y in range(len(list)-x-1): #Length of array=10, initial value of x = 0. 10-0-9 = 1. Traverse 0 to 1
                        if (list[y] > list[y+1]):
                                list[y], list[y+1] = list[y+1], list[y]
        return list

arrlist = [67,90,0,-40,9,192,333,3892,-22,-434] #sample array to be sorted
bubbleSort(arrlist)
print("Sorted list: ",arrlist)
